dffml/source/mysql_source.py

```diff
@@ -0,0 +1,202 @@
+from collections import OrderedDict
+
+from dffml.base import BaseConfig
+from dffml.repo import Repo
+from dffml.source.source import BaseSourceContext, BaseSource
+from dffml.util.cli.arg import Arg
+from dffml.util.entrypoint import entry_point
+
+
+class MysqlSourceConfig(BaseConfig, NamedTuple):
+    host: str
+    port: int
+    user: str
+    password: str
+    db: str
+    table: str
+    update_query: str
+    repos_query: str
+    repo_query: str
+    model_columns: str
+
+
+class MysqlSourceContext(BaseSourceContext):
+    async def update(self, repo: Repo):
+        update_query = self.parent.config.update_query
+        if update_query == "":
+            update_query = "insert into " + self.parent.config.table + "("
+            values = "("
+            column_value_pair = OrderedDict()
+            for column_name in self.parent.config.model_columns.split():
+                if column_name.startswith("feature_"):
+                    key = column_name.replace("feature_", "")
+                    if key in repo.data.features.keys():
+                        values += str(repo.data.features[key]) + ","
+                        column_value_pair[column_name] = repo.data.features[
+                            key
+                        ]
+                        update_query += column_name + ","
+                elif column_name.startswith("prediction_"):
+                    key = column_name.replace("prediction_", "")
+                    if (
+                        "prediction" in repo.data.__dict__.keys()
+                        and key in repo.data.prediction.keys()
+                    ):
+                        if key == "value":
+                            values += (
+                                "'" + str(repo.data.prediction[key]) + "',"
+                            )
+                            column_value_pair[column_name] = (
+                                "'" + str(repo.data.prediction[key]) + "'"
+                            )
+                        else:
+                            values += str(repo.data.prediction[key]) + ","
+                            column_value_pair[
+                                column_name
+                            ] = repo.data.prediction[key]
+                        update_query += column_name + ","
+                else:
+                    key = column_name
+                    if key in repo.data.__dict__.keys():
+                        values += repo.data.__dict__[key] + ","
+                        column_value_pair[column_name] = repo.data.__dict__[
+                            key
+                        ]
+                        update_query += column_name + ","
+            update_query = update_query[:-1] + ")"
+            values = values[:-1] + ")"
+            update_query += " values " + values + "on duplicate key update "
+            for key, value in column_value_pair.items():
+                update_query += key + "=" + str(value) + ","
```

Comment:
Building SQL queries can be dangerous, see [SQL Injection](https://www.w3schools.com/sql/sql_injection.asp) for more info. The big thing to learn about SQL injection is that you need to bind your parameters. 

In the following example `feature_cols` is your `model_columns`.

https://github.com/intel/dffml/blob/86a0168fdd42e59fee427bb6569ed81cd12bb9e0/examples/source/custom_sqlite.py#L19-L28

However sqlite uses `?` to bind parameters, where as `aiomysql` uses `%s`

https://github.com/intel/dffml/blob/86a0168fdd42e59fee427bb6569ed81cd12bb9e0/examples/maintained/demoapp/source.py#L29-L33

The last argument to the `execute` call is a `tuple` of parameters which are associated with each `%s`, similar to string formatting.